### ISSUE TITLE
MP4ThumbsAndGif replaces MP3 file posters

### ISSUE THREAD

**Issue by hervaden** [WWBN/AVideo#8119]
**Describe the bug**
When MP4ThumbsAndGif plugin is switched off, and then reenabled after the upload of an MP3 file (with thumbnail), it triggers and replaces the uploaded thumbnail with a generated "wave form" from FFMPEG

**To Reproduce**
1. Install the MP4ThumbAndGif plugin
2. Use it as regular
3. Disable the plugin
4. Upload an MP3 file
5. Add a poster to the uploaded file
6. Reenable the plugin
7. Edit or view an MP3 itme
8. The uploaded thumbnail will be replaced by the generated wavefrom 

**Expected behavior**
The plugin should honour the uploaded poster and only generate an automated poster if there is no poster present

**Screenshots**
Attached

**Desktop (please complete the following information):**
 - OS: MacOS Ventura / LINUX Ubuntu
 - Browser: Chrome
 - Version: Latest and patched

**Smartphone (please complete the following information):**
N/A

**Additional context**
Have not updated to the latest plugin version because it is incompatible with the one version back of the AVideo system that is used.

![Screenshot 2023-06-21 at 7 54 47 AM](https://github.com/WWBN/AVideo/assets/59710655/f95be29b-8f2f-4d3f-b81f-84de8364e0b2)
![Screenshot 2023-06-21 at 6 59 07 AM](https://github.com/WWBN/AVideo/assets/59710655/871c63f6-7ddd-45d0-9636-17b0c5855cc6)
![Screenshot_20230621_065646](https://github.com/WWBN/AVideo/assets/59710655/09233547-8357-44e5-9265-1a9eedf5fb1f)
![Screenshot 2023-06-21 at 6 55 17 AM](https://github.com/WWBN/AVideo/assets/59710655/3c852288-5101-437e-ac16-a851ec0194d7)
![Screenshot 2023-06-21 at 6 54 57 AM](https://github.com/WWBN/AVideo/assets/59710655/e1213adf-5c42-40b2-8fe4-f6a51aa8f2a6)
![Screenshot 2023-06-21 at 6 54 36 AM](https://github.com/WWBN/AVideo/assets/59710655/aedd062b-b09c-4824-8fc0-bf9fba19f6ef)

**Comment by DanielnetoDotCom** [comment]
I will make a new version with an option to disable extract the spectrum image from MP3 

but it will require you to update your AVideo, we do not support older versions.

**Comment by hervaden** [comment]
Thank you.  I cannot update AVideo.  Could you indicate the files I need to update in AVideo to not allow the plugin to cause errors?

**Comment by DanielnetoDotCom** [comment]
the plugin file is encrypted you will not be able to modify it, also the reason why I do not support older versions is because is impossible to update only part of the code. it really requires you to update everything.

**Comment by DanielnetoDotCom** [comment]
BTW, in the MP4ThumbsAndGif plugin version 3.7, you will find an option to disable the MP3 spectrum generation

**Comment by hervaden** [comment]
I am looking for the functions it is complaining about not getting in the portion where the code is open.  It seems as if there is a function added in the latest AVideo that is not in the one before.  if I could just update that file to ensure the plugin complies I would appreciate it.

**Comment by DanielnetoDotCom** [comment]
I understand your situation, however, I'm unable to provide guidance or solutions specifically for older versions of AVideo. As previously stated, supporting these versions is not feasible due to their complex, intertwined codebase. It's essential to update AVideo entirely for the best performance and compatibility. Thank you for your understanding.

**Comment by hervaden** [comment]
Traced the error:

[23-Jun-2023 11:44:07 America/Vancouver] AVideoLog::ERROR: {"type":1,"message":"Uncaught Error: Call to undefined function url_exists() in \/mnt\/volume_tor1_01\/www\/tcs\/plugin\/MP4ThumbsAndGif\/MP4ThumbsAndGif.php(18) : eval()'d code(1) : eval()'d code:85\nStack trace:\n#0 \/mnt\/volume_tor1_01\/www\/tcs\/view\/mini-upload-form\/upload.php(122): MP4ThumbsAndGif::getImage()\n#1 {main}\n  thrown","file":"\/mnt\/volume_tor1_01\/www\/tcs\/plugin\/MP4ThumbsAndGif\/MP4ThumbsAndGif.php(18) : eval()'d code(1) : eval()'d code","line":85} SCRIPT_NAME: /view/mini-upload-form/upload.php


and

[23-Jun-2023 11:48:06 America/Vancouver] AVideoLog::ERROR: {"type":1,"message":"Uncaught Error: Call to undefined method Video::forceAudio() in \/mnt\/volume_tor1_01\/www\/tcs\/plugin\/MP4ThumbsAndGif\/MP4ThumbsAndGif.php(18) : eval()'d code(1) : eval()'d code:235\nStack trace:\n#0 \/mnt\/volume_tor1_01\/www\/tcs\/plugin\/MP4ThumbsAndGif\/MP4ThumbsAndGif.php(18) : eval()'d code(1) : eval()'d code(410): MP4ThumbsAndGif::getImageInDuration()\n#1 \/mnt\/volume_tor1_01\/www\/tcs\/plugin\/AVideoPlugin.php(792): MP4ThumbsAndGif->onUploadIsDone()\n#2 \/mnt\/volume_tor1_01\/www\/tcs\/view\/mini-upload-form\/upload.php(149): AVideoPlugin::onUploadIsDone()\n#3 {main}\n  thrown","file":"\/mnt\/volume_tor1_01\/www\/tcs\/plugin\/MP4ThumbsAndGif\/MP4ThumbsAndGif.php(18) : eval()'d code(1) : eval()'d code","line":235} SCRIPT_NAME: /view/mini-upload-form/upload.php


Added the following code to satisfy the need:


objects/functions.php:

function url_exists($url) {
    global $global;
    if (preg_match('/^https?:\/\//i', $url)) {
        $parts = explode('/videos/', $url);
        if (!empty($parts[1])) {
            $tryFile = "{$global['systemRootPath']}videos/{$parts[1]}";
            //_error_log("try_get_contents_from_local {$url} => {$tryFile}");
            if (file_exists($tryFile)) {
                return $tryFile;
            }
        }
        $file_headers = get_headers($url);
        if (empty($file_headers)) {
            _error_log("url_exists($url) empty headers");
            return false;
        } else {
            foreach ($file_headers as $value) {
                if (preg_match('/404 Not Found/i', $value)) {
                    _error_log("url_exists($url) 404 {$value}");
                    return false;
                }
            }
            return true;
        }
    } else {
        $exists = file_exists($url);
        if ($exists == false) {
            _error_log("url_exists($url) local file do not exists");
        }
        return $exists;
    }
}

objects/video.php:

public static function forceAudio() {
            if (!empty($_REQUEST['includeType'])) {
                if ($_REQUEST['includeType'] === 'audio') {
                    return true;
                }
            }
            return false;
        }

        public static function forceArticle() {
            if (!empty($_REQUEST['includeType'])) {
                if ($_REQUEST['includeType'] === 'article') {
                    return true;
                }
            }
            return false;
        }




In short:
replacing these two files with the latest should resolve te issue.

**Comment by hervaden** [comment]
@DanielnetoDotCom Even with the new component, the problem persists.  The switch is off for mp3, but it still removes my poster.

**Comment by hervaden** [comment]
Extra note: this kicks off on view, not just on edit.
